server/routes/unusedDeductionRoutes.ts

```diff
@@ -96,6 +140,38 @@ export default class UnusedDeductionRoutes {
     const { bookingId, prisonerNumber } = res.locals.prisoner
     const { username } = res.locals.user
     const adjustments = await this.getAdjustments(bookingId, username, nomsId)
+    const reviewDeductions = this.paramStoreService.get(req, 'returnToReviewDeductions')
+    if (reviewDeductions) {
+      const sessionAdjustmentRecords = this.adjustmentsStoreService.getAll(req, nomsId)
+      const sessionAdjustments: SessionAdjustment[] = Object.keys(sessionAdjustmentRecords).map(
+        key => ({ ...sessionAdjustmentRecords[key] }) as SessionAdjustment,
+      )
+
+      adjustments
+        .filter(it => (it.adjustmentType === 'REMAND' || it.adjustmentType === 'TAGGED_BAIL') && it.source === 'NOMIS')
+        .forEach(it => {
+          if (!sessionAdjustments.find(adj => adj.id === it.id)) {
+            sessionAdjustments.push(it)
+          }
+        })
+
+      const observables: Promise<void | {
+        adjustmentIds: string[]
+      }>[] = []
+      sessionAdjustments.forEach(it => {
+        if (it.delete) {
+          observables.push(this.adjustmentsService.delete(it.id, username))
+        } else if (this.paramStoreService.get(req, it.id)) {
+          observables.push(this.adjustmentsService.create([it], username))
+        } else {
+          observables.push(this.adjustmentsService.update(it.id, it, username))
+        }
+      })
+
+      await Promise.all(observables)
+      return res.redirect(`/${nomsId}/success`)
+    }
+
     const unusedDeductionAdjustment = this.getUnusedDeduction(adjustments)
```

Comment:
Can you split this into a separate endpoint? Seems odd to have MUD + RUD go through the same endpoints.